doc/flakes.adoc

```diff
@@ -1,2 +1,119 @@
 [[ch-flakes]]
 == Using Home Manager with Flakes
```

Comment:
Since we support both using this with NixOS and Generic Linux as well as macOS. Should we add a sub-header to mention this is for NixOS integration?